src/timeline/timeline.style.tsx

```diff
@@ -1,60 +1,96 @@
 import styled, { css } from "styled-components";
+import { TickIcon } from "@lifesg/react-icons";
 import { Color } from "../color";
 import { MediaQuery } from "../media";
 import { Text } from "../text";
-import { TimelineStatusProps } from "./types";
+import { TimelineStatusProps, Variant } from "./types";
 
 // =============================================================================
 // STYLE INTERFACES, transient props are denoted with $
 // See more https://styled-components.com/docs/api#transient-props
 // =============================================================================
 
-interface CircleIndicatorStyleProps {
-    $filled?: boolean | undefined;
-}
-
 interface TimelineWrapperStyleProps {
     $startCol?: number | undefined;
     $colSpan?: number | undefined;
 }
 
+interface VariantStyleProps {
+    $variant: Variant;
+}
+
 // =============================================================================
 // STYLE INTERFACES
 // =============================================================================
-
-export const CircleIndicator = styled.div<CircleIndicatorStyleProps>`
-    width: 1rem;
-    height: 1rem;
-    margin-top: 0.5rem;
+export const CircleIndicator = styled.div<VariantStyleProps>`
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: 1.5rem;
+    height: 1.5rem;
+    margin-top: 4px;
     border-radius: 50%;
+
     ${(props) => {
-        if (props.$filled) {
-            return css`
-                border: none;
-                background-color: ${Color.Accent.Light[1]};
-            `;
+        switch (props.$variant) {
+            case "current":
+                return css`
+                    background-color: ${Color.Accent.Light[1]};
+                `;
+            case "upcoming-active":
+                return css`
+                    border: 4px solid ${Color.Accent.Light[1]};
+                `;
+            case "upcoming-inactive":
+                return css`
+                    border: 4px solid ${Color.Neutral[4]};
+                `;
+            case "completed":
+                return css`
+                    background-color: ${Color.Validation.Green.Icon};
+                    svg {
+                        display: block;
+                    }
+                `;
         }
-
-        return css`
-            border: 3.2px solid ${Color.Accent.Light[1]};
-            background-color: transparent;
-        `;
     }}
 `;
 
-export const LineIndicator = styled.div`
-    width: 0.25rem;
+export const IconTick = styled(TickIcon)`
+    color: ${Color.Neutral[8]};
+    display: none;
+`;
+
+export const LineIndicator = styled.div<VariantStyleProps>`
+    width: 4px;
```

Comment:
did you mean to change this to px?

Reply:
ya. i thoughts we will use `px` if the value is small enough. haha

Reply:
hmm I'd say usually spacing can be kept in rem even for 4px/8px

but in this case it needs to follow the border width which is in px, so yea this is fine